src/basic/list.h

```diff
@@ -190,3 +188,5 @@
                         LIST_REMOVE(name, *_a, _p);                     \
                 _p;                                                     \
         })
+
+#include "macro.h"
```

Comment:
hmm, this feels like a hack, but i got no better idea atm.

But please add a comment why this is needed. i.e something like this:

```c
/* Now include "macro.h", because we want our definition of assert() which the macros above use. We include it down here instead of up top, since macro.h pulls in log.h which in turn needs our own definitions. */
```

or something like that

Reply:
(BTW, we already do that in macro.h, i.e. log.h is included at the end of macro.h)